### ISSUE TITLE
Constrain image height of galleries

### ISSUE THREAD

**Issue by jaredh159** [netrivet/prophoto-issues#627]
Probably an opt-in feature for each gallery style would work best.

**Comment by jaredh159** [comment]
+1 Steve Bennet 143771

**Comment by postmastersteve** [comment]
+1 Gustavo Gaiote ticket 143892

**Comment by benjamintennant** [comment]
+1 Kristin 3524

**Comment by benjamintennant** [comment]
+1 Jenny Myers from feedback

**Comment by postmastersteve** [comment]
+1 Alexandra Ryoo ticket 144359

**Comment by postmastersteve** [comment]
+1 Cheryl ticket 3664

**Comment by danjonlam** [comment]
+1 Heather Carson 3714

**Comment by postmastersteve** [comment]
+1 Lisa Jones ticket 144815

**Comment by postmastersteve** [comment]
+1 Susan Dodd ticket 3914

**Comment by postmastersteve** [comment]
+1 Jennifer Wolters ticket 145141

**Comment by danjonlam** [comment]
+1 Charles Coons ticket 145397

**Comment by postmastersteve** [comment]
+1 David Brown ticket 146031

**Comment by danjonlam** [comment]
+1 Ester Ammerlaan 146225

**Comment by postmastersteve** [comment]
+1 Nely Ariza ticket 146229

**Comment by postmastersteve** [comment]
+1 Angela King ticket 146373

**Comment by postmastersteve** [comment]
+1 Trish Reda ticket 146160

**Comment by danjonlam** [comment]
+1 Constance Schiano ticket 146751

**Comment by danjonlam** [comment]
+1 Jamie Redmond ticket 145692 - specifically for the thumbnail gallery grid style, when the images appear in the lightbox display

**Comment by postmastersteve** [comment]
+1 Bill Irwin 146958

**Comment by danjonlam** [comment]
+1 Gina Britton ticket 147151

**Comment by postmastersteve** [comment]
+1 Angela Navarette
Thumbnail gallery full-browser view constraint (pixel limit? screen percentage limit?)

**Comment by postmastersteve** [comment]
+1 Sebastian Gonzalez ticket 147593

**Comment by postmastersteve** [comment]
+1 Alexandra Minton ticket 147890

**Comment by postmastersteve** [comment]
+1 Amy Osborne ticket 148020

**Comment by postmastersteve** [comment]
+1 Hannah Whaley ticket 148270

**Comment by postmastersteve** [comment]
+1 Melody Taylor ticket 148651

**Comment by postmastersteve** [comment]
+1 Cathy Murai ticket 149095

**Comment by postmastersteve** [comment]
+1 Pim Yanaprasart ticket 149092

**Comment by benjamintennant** [comment]
+1 Erika Blunt ticket 149355

**Comment by benjamintennant** [comment]
+1 Nicole Langen ticket 149467

**Comment by danjonlam** [comment]
+1 Jamie Redmond ticket 149545

**Comment by danjonlam** [comment]
+1 Tiffany Salienko ticket 149709

**Comment by benjamintennant** [comment]
+1 Grayson Smith ticket 149744

**Comment by postmastersteve** [comment]
+1 Salimah Vaiya ticket 150055

**Comment by postmastersteve** [comment]
+1 Michelle Heron ticket 150358

**Comment by postmastersteve** [comment]
@brianium @jaredh159 super-stupid question ... is there any reason this isn't a good solution until we can provide a more fleshed-out UI for controlling & limiting the height of P6 galleries?

```
.ansel-slides-container, .ansel-slides-container .slide img {
    max-height: 100vh;
    width: auto !important;
}
```

So many people have asked for this kind of default behavior, so I ran a quick test with that code on one person's slideshow gallery and it seems to work ... maybe I'm missing a condition where that behavior is undesirable, though...

**Comment by benjamintennant** [comment]
+1 Jay Rathbun ticket 150475

**Comment by benjamintennant** [comment]
+1 Joni Bilderback ticket 150476

**Comment by postmastersteve** [comment]
+1 Laura Mesarec ticket 150456

**Comment by benjamintennant** [comment]
+1 Janine Guidera ticket 150787

**Comment by postmastersteve** [comment]
+1 Jaylene Gepford ticket 150811

**Comment by benjamintennant** [comment]
+1 Sharon Gerschwitz ticket 150862

**Comment by danjonlam** [comment]
+1 Rosa Tseng ticket 150894

**Comment by postmastersteve** [comment]
+1 Stephanie Richartz ticket 150915

**Comment by postmastersteve** [comment]
+1 Laura Mesarec ticket 150456

**Comment by danjonlam** [comment]
+1 Duke Seigars ticket 150925

**Comment by danjonlam** [comment]
+1 Jarosław Tomaszewski ticket 151195

**Comment by postmastersteve** [comment]
+1 Sam Rummans ticket 151213

**Comment by postmastersteve** [comment]
+1 Kathy Costello ticket 151328

**Comment by benjamintennant** [comment]
+1 Fanny Auer ticket 151448

**Comment by danjonlam** [comment]
+1 kdgoldie@gmail.com ticket 151484

**Comment by benjamintennant** [comment]
+1 Melissa ticket 150540

**Comment by benjamintennant** [comment]
+1 ‎Camryn Shonka Facebook support group

**Comment by postmastersteve** [comment]
+1 Amy Osborne ticket 151933

**Comment by postmastersteve** [comment]
+1 Lori Thompson ticket 152111

**Comment by postmastersteve** [comment]
+1 Elias Kordelakos ticket 152114

**Comment by danjonlam** [comment]
+1 Megan Simpson ticket 152231

**Comment by postmastersteve** [comment]
+1 Leslie Halper ticket 152339

**Comment by benjamintennant** [comment]
+1 Hal Haygood ticket 152341

**Comment by postmastersteve** [comment]
+1 Rachel Mord ticket 152463

**Comment by danjonlam** [comment]
+1 Chris Reichman ticket 152492

**Comment by danjonlam** [comment]
+1 Ann Pak ticket 153492

**Comment by benjamintennant** [comment]
+1 Becky Shivers ticket 153729

**Comment by postmastersteve** [comment]
+1 Sally Salerno ticket 153950

**Comment by postmastersteve** [comment]
+1 Samantha Petrushak ticket 154075

**Comment by postmastersteve** [comment]
+1 Francine Mittenthal ticket 154500

**Comment by benjamintennant** [comment]
+1 Tony Kim ticket 154599

**Comment by postmastersteve** [comment]
+1 Joice Ruffing ticket 153832

**Comment by postmastersteve** [comment]
+1 Andrea Zajonc ticket 155091

**Comment by postmastersteve** [comment]
+1 Joe ticket 155380

**Comment by benjamintennant** [comment]
+1 Teo ticket 155443 wants to be able to constrain size of expanded mode image, at least to a maximum of the original size.

**Comment by danjonlam** [comment]
+1 Anna Dziwak ticket 155629

**Comment by postmastersteve** [comment]
+1 Audrée Wilhelmy ticket 155842

**Comment by danjonlam** [comment]
+1 Viktor Isaak ticket 155956 - prevent image upscaling

**Comment by danjonlam** [comment]
+1 Marije Van Der Leeuw ticket 155679

**Comment by danjonlam** [comment]
+1 Kueste Kreativ Tania und Chris Manteufel ticket 156517

**Comment by postmastersteve** [comment]
+1 Marcus ticket 155499

**Comment by benjamintennant** [comment]
+1 Marion Dmnt ticket 156865

**Comment by postmastersteve** [comment]
+1 Benjamin Thorn ticket 157019

**Comment by postmastersteve** [comment]
+1 Erin Stevens ticket 157175

**Comment by postmastersteve** [comment]
+1 Susy Shearer ticket 157291

**Comment by postmastersteve** [comment]
+1 Susan Adams ticket 157406

**Comment by postmastersteve** [comment]
+1 Lauren McCormick ticket 157378

**Comment by postmastersteve** [comment]
+1 Piotr Dziwak ticket 157438

**Comment by postmastersteve** [comment]
+1 Ashley Warren ticket 157439

**Comment by postmastersteve** [comment]
+1 Benny ticket 157492

**Comment by danjonlam** [comment]
+1 Gus Cutting ticket 157728

**Comment by postmastersteve** [comment]
+1 John Croxton ticket 157324

**Comment by danjonlam** [comment]
+1 Andrea ticket 157972

**Comment by benjamintennant** [comment]
+1 Lee Wen Cheng ticket 158234

**Comment by benjamintennant** [comment]
+1 Jennifer McCarty ticket 158242 - just wants smaller images in expanded mode

**Comment by danjonlam** [comment]
+1 Alex Bradbury ticket 158442

**Comment by danjonlam** [comment]
+1 Dana Moran ticket 158443

**Comment by benjamintennant** [comment]
+1 Stacy Balfour ticket 158504

**Comment by benjamintennant** [comment]
+1 Anna Zeiter ticket 158233

**Comment by jaredh159** [comment]
Closing because fixed/added in [6.16.0](https://pro.photo/blog/prophoto-6-16-0) on 2/27/17.
